### ISSUE TITLE
Describe infinite loop in Operators and Control Flow section

### ISSUE THREAD

**Issue by Beliavsky** [fortran-lang/fortran-lang.org#271]
The Operators and Control Flow [section](https://fortran-lang.org/learn/quickstart/operators_control_flow) presents the do while loop with the example
```
  integer :: i
  i = 1
  do while (i < 11)
    print *, i
    i = i + 1
  end do
  ! Here i = 11
```
but the infinite do loop can emulate this behavior and is more general, since you can test the condition anywhere in the loop. I use do loops with indices and infinite do loops with exits, but not do while. So I suggest that the infinite do loop be presented in addition to or instead of do while. The equivalent to the code above would be

```
  integer :: i
  i = 1
  do
    if (i > 10) exit
    print *, i
    i = i + 1
  end do
  ! Here i = 11
  print*,i
  end
```
The code uses the one-line-if, which I suggested mentioning in another issue.

**Comment by LKedward** [comment]
Yes we should present the option of an 'infinite' `do` loop with a conditional `exit` __in addition__ to the existing loop examples. Rather than replacing the existing `do while` example, I suggest we simply adapt the first excerpt in the [Loop control statements](https://github.com/fortran-lang/fortran-lang.org/blob/master/learn/quickstart/operators_control_flow.md#loop-control-statements-exit-and-cycle) section to be an 'infinite' loop and add text to explain the lack of counter variable as a valid construct.

As you allude to, the choice of an 'infinite' `do` loop versus a `do while` loop is merely a personal choice and so we should objectively present both options. (I personally prefer `do while` loops when there is only one terminating condition since it makes the exit condition clear upfront without needing to inspect the body of the loop.)

**Comment by arjenmarkus** [comment]
Would it be an idea to gather typical constructions (the "Fortran idiom")?
Mind you, I have not looked in detail at the tutorial, just something I
realised might be useful.

Op di 25 mei 2021 om 14:39 schreef Laurence Kedward <
***@***.***>:
